aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -138,241 +365,225 @@ def export(
         raise exceptions.ArchiveExportError(f"The output file '{filename}' already exists")
 
     if silent:
-        logging.disable(level=logging.CRITICAL)
+        logging.disable(logging.CRITICAL)
+
+    writer = writers[file_format](filename=filename, use_compression=use_compression)
+
+    summary(
+        writer.file_format_verbose,
+        filename,
+        include_comments=include_comments,
+        include_logs=include_logs,
+        **traversal_rules
+    )
+
+    output_data = {}
+
+    output_data['time_extract_start'] = time.time()
+    export_data = _generate_data(
+        entities=entities,
+        silent=silent,
+        allowed_licenses=allowed_licenses,
+        forbidden_licenses=forbidden_licenses,
+        include_comments=include_comments,
+        include_logs=include_logs,
+        **traversal_rules
+    )
+    output_data['time_extract_stop'] = time.time()
+
+    extract_time = output_data['time_extract_start'] - output_data['time_extract_stop']
+    EXPORT_LOGGER.debug(f'Data extracted in {extract_time:6.2g} s.')
+
+    if export_data is not None:
+        try:
+            output_data['time_write_start'] = time.time()
+            output_data['writer'] = writer.write(export_data=export_data, silent=silent)  # type: ignore
```

Comment:
Yeh I just added this part, mainly so it was compatible with `export_tar` (to get the compression timings). Really it could return anything that would be helpful for e.g. performance analysis
My thinking was not to merge directly, just in case the writer data overwrote the "base" keys.
In fact `output_data` could also be turned into a dataclass

Reply:
dataclass really sounds like the right approach here